lib/internal/process/warning.js

```diff
@@ -41,9 +49,15 @@ function setupProcessWarnings() {
     if (!(warning instanceof Error)) {
       throw new TypeError('\'warning\' must be an Error object or string.');
     }
-    if (throwDeprecation && warning.name === 'DeprecationWarning')
+    if (throwDeprecation && warning.name === 'DeprecationWarning') {
       throw warning;
-    else
-      process.nextTick(() => process.emit('warning', warning));
+    } else {
+      if (!warned) {
```

Comment:
Upd: after the last update, this is not a lonely if anymore, so disregard this comment.